gprofiler/profilers/perf.py

```diff
@@ -397,20 +404,21 @@ def __init__(
                 os.path.join(self._profiler_state.storage_dir, "perf.fp"),
                 False,
                 perf_inject,
-                [],
+                extra_args,
             )
             self._perfs.append(self._perf_fp)
         else:
             self._perf_fp = None
 
         if perf_mode in ("dwarf", "smart"):
+            extra_args.extend(["--call-graph", f"dwarf,{perf_dwarf_stack_size}"])
             self._perf_dwarf: Optional[PerfProcess] = PerfProcess(
                 self._frequency,
                 self._profiler_state.stop_event,
                 os.path.join(self._profiler_state.storage_dir, "perf.dwarf"),
                 True,
                 False,  # no inject in dwarf mode, yet
-                ["--call-graph", f"dwarf,{perf_dwarf_stack_size}"],
+                extra_args,
```

Comment:
```suggestion
                extra_args + ["--call-graph", f"dwarf,{perf_dwarf_stack_size}"],
```
Since extra_args is used for other purposes in this function, editing it with `extend` will affect code that we might add later in the function... Making this piece more fragile. In such cases it's preferable not to edit